### PR TITLE
Walker fix and less verbose form

### PR DESCRIPTION
**Pull request by ikitommi**
* less verbose spec form
* update deps
* better spec walker
* `st/select-spec` uses the walker, from Nekala!
* composable type transformers!

### CODE REVIEW

File: src/spec_tools/transform.cljc [hunk 1]
```diff
@@ -117,6 +117,16 @@
     ::s/invalid
     x))
 
+;;
+;; Tuples
+;;
+
+(defn strip-extra-values [{:keys [::parse/items]} x]
+  (let [size (and (count items))]
```
Comment (kalekale) [inline src/spec_tools/transform.cljc]: `and` what?

File: src/spec_tools/core.cljc [hunk 2]
```diff
@@ -235,20 +246,23 @@
...
-           (= (count value)
-              (count items)))
-    (let [pairs (map vector value items)]
-     (map (fn [[v item]]
-            (accept item v options))
-          pairs))
+  (if (sequential? value)
+    (->> (map-indexed vector value)
+         (map (fn [[i v]]
+                (if (if (< i (count items)) (nth items i))
```
[the remark this answers is not in the document]
Comment (ikitommi) [reply src/spec_tools/core.cljc]:
```clj
(fn [[i v]]
  (if (< i (count items))
    (some-> (nth items i) (accept v options))
    v))
```

File: src/spec_tools/core.cljc [hunk 3]
```diff
@@ -235,20 +246,23 @@
...
 (defmethod walk :tuple [{:keys [::parse/items]} value accept options]
-  (if (and (sequential? value)
-           (= (count value)
-              (count items)))
-    (let [pairs (map vector value items)]
-     (map (fn [[v item]]
-            (accept item v options))
-          pairs))
+  (if (sequential? value)
+    (->> (map-indexed vector value)
```
[the remark this answers is not in the document]
Comment (ikitommi) [reply src/spec_tools/core.cljc]: 👍

File: src/spec_tools/core.cljc [hunk 4]
```diff
@@ -81,24 +81,31 @@
...
-        (or (get spec encode-key)
-            (get encoders (:type spec))
-            default-encoder))
-      (-decoder [_ spec _]
-        (or (get spec decode-key)
-            (get decoders (:type spec))
-            default-decoder)))))
+(defn type-transformer
+  "Returns a Transformer instance out of options or Transformer instances.
+  Merges "
```
[the remark this answers is not in the document]
Comment (ikitommi) [reply src/spec_tools/core.cljc]: made better docs

File: src/spec_tools/transform.cljc (hunk 1 above)
Comment (ikitommi) [reply src/spec_tools/transform.cljc]: ugh.

**Comment by ikitommi** [comment]
Fixed all of those, thanks!

**Pull request merged**